### ISSUE TITLE
Windows shouldn't allow IME events when not editing text (On Desktop)

### ISSUE THREAD

**Issue by liquidhelium** [emilk/egui#2614]
**Describe the bug**
We can always enable or disable IME because IME is always allowed. In a typical application, this attribute should be determined by whether a text is being edited.
https://github.com/emilk/egui/blob/0eabd894bd0e4466d53e3098e8cce1f10183f3b0/crates/eframe/src/native/run.rs#L563-L572
https://github.com/emilk/egui/blob/0eabd894bd0e4466d53e3098e8cce1f10183f3b0/crates/eframe/src/native/run.rs#L1011-L1024

**To Reproduce**
1. You'll need to have a input method editor.
2. Start a *native* egui app (on web it works fine) and you'll find that you can call up the IME when not editing (even there's no `textedit`).

**Expected behavior**
When editing, IME is allowed; Disallowed otherwise.

**Screenshots**
![IME is on it's initial position.](https://user-images.githubusercontent.com/59129152/213909537-08972c86-0a54-4d72-a758-c09803896370.png)
(IME is on it's initial position.)

**Desktop (please complete the following information):**
 - OS: Windows 22H2 (Windows 11)

**Additional context**
There's a field `mutable_text_under_cursor` in [`egui::output::PlatformOutput`](https://docs.rs/egui/latest/egui/output/struct.PlatformOutput.html).

currently it's used by `eframe` web, but it should also applicable to `egui_winit` or other crates.

**Comment by Barugon** [comment]
if `[gl_]window.set_ime_allowed` was called with true when a text `edit` gained focus and called with `false` when that text edit lost focus then the on-screen keyboard would properly show and hide on Linux devices like PinePhone and Librem 5.
